includes/payment-methods/class-wc-stripe-upe-payment-method-bacs-debit.php

```diff
@@ -93,22 +96,64 @@ public function create_payment_token_for_user( $user_id, $payment_method ) {
 		return $token;
 	}
 
-	public function hide_bacs_for_subscriptions_with_free_trials() {
+	/**
+	 * Conditionally hides the Bacs payment gateway for specific scenarios.
+	 */
+	public function maybe_hide_bacs_payment_gateway() {
 		add_filter(
 			'woocommerce_available_payment_gateways',
 			function ( $available_gateways ) {
-				global $post;
-				$is_checkout_shortcode_page          = wc_post_content_has_shortcode( 'woocommerce_checkout' ) || has_block( 'woocommerce/classic-shortcode', $post );
-				$is_update_order_review_ajax_request = defined( 'DOING_AJAX' ) && DOING_AJAX && isset( $_REQUEST['wc-ajax'] ) && 'update_order_review' === $_REQUEST['wc-ajax'];
-				if ( $is_checkout_shortcode_page || $is_update_order_review_ajax_request ) {
-					// Checking if the amount is zero allows us to process orders that include subscriptions with a free trial,
-					// as long as another product increases the total amount, ensuring compatibility with Bacs.
-					if ( class_exists( 'WC_Subscriptions_Cart' ) && WC_Subscriptions_Cart::cart_contains_free_trial() && (float) WC()->cart->total === 0.00 ) {
-						unset( $available_gateways['stripe_bacs_debit'] );
-					}
+				if (
+				$this->should_hide_bacs_for_pre_orders_charge_upon_release() ||
+				$this->should_hide_bacs_for_subscriptions_with_free_trials()
```

Comment:
Nit: Should these be aligned with the if-then parentheses?

```suggestion
					$this->should_hide_bacs_for_pre_orders_charge_upon_release() ||
					$this->should_hide_bacs_for_subscriptions_with_free_trials()
```